crates/wasm/tests/test_build.rs

```diff
@@ -0,0 +1,469 @@
+        let full_viewing_key = "penumbrafullviewingkey1mnm04x7yx5tyznswlp0sxs8nsxtgxr9p98dp0msuek8fzxuknuzawjpct8zdevcvm3tsph0wvsuw33x2q42e7sf29q904hwerma8xzgrxsgq2";
+
+        // Serialize transaction plan into `TransactionPlan`.
+        let transaction_plan_serialized: tp =
+            serde_wasm_bindgen::from_value(transaction_plan.clone()).unwrap();
+        let transaction_plan_conv: TransactionPlan =
+            transaction_plan_serialized.try_into().unwrap();
+
+        // -------------- 4. Build the (1) Serial Transaction and (2) Parallel Transaction --------------
+
+        let mut actions: Vec<Action> = Vec::new();
+
+        for i in transaction_plan_conv.actions.clone() {
+            if let ActionPlan::Spend(ref _spend_plan) = i {
+                let action_deserialize = serde_wasm_bindgen::to_value(&i).unwrap();
+                let action = wasm_planner
+                    .build_action(
+                        transaction_plan.clone(),
+                        action_deserialize,
+                        full_viewing_key,
+                        witness_data.as_ref().unwrap().clone(),
+                    )
+                    .unwrap();
+                let action_serialize: Action =
+                    serde_wasm_bindgen::from_value(action.clone()).unwrap();
+                actions.push(action_serialize);
+            }
+        }
+
+        for i in transaction_plan_conv.actions {
+            if let ActionPlan::Output(ref _output_plan) = i {
+                let action_deserialize = serde_wasm_bindgen::to_value(&i).unwrap();
+                let action = wasm_planner
+                    .build_action(
+                        transaction_plan.clone(),
+                        action_deserialize,
+                        full_viewing_key,
+                        witness_data.as_ref().unwrap().clone(),
+                    )
+                    .unwrap();
+                let action_serialize: Action =
+                    serde_wasm_bindgen::from_value(action.clone()).unwrap();
+                actions.push(action_serialize);
+            }
+        }
+
+        // Deserialize actions.
+        let action_deserialized = serde_wasm_bindgen::to_value(&actions).unwrap();
+
+        // Execute parallel spend transaction and generate proof.
+        let parallel_transaction = build_parallel(
+            action_deserialized,
+            transaction_plan.clone(),
+            witness_data.as_ref().unwrap().clone(),
+            authorization_data.clone(),
+        )
+        .unwrap();
+        console_log!("Parallel transaction is: {:?}", parallel_transaction);
+
+        let serial_transaction = build(
+            full_viewing_key,
+            transaction_plan.clone(),
+            witness_data.as_ref().unwrap().clone(),
+            authorization_data.clone(),
+        )
+        .unwrap();
+        console_log!("Serial transaction is: {:?}", serial_transaction);
+    }
+}
```

Comment:
I didn't see any assertions in the test, should we add some?

Reply:
The assertions here would be tough because we're essentially comparing if a serial transaction and parallel transaction match. The only difference in their transaction payload should be the binding signature since it's based on randomness. I've been instead manually inspecting if they match in the dev-console, but some kind of assertion would be nice.